File: src/co/soundwave/repository/artist/ArtistSQL.kt
package co.soundwave.repository.artist

import co.soundwave.repository.BaseSQL

class ArtistSQL : BaseSQL() {
    fun insertArtist(name: String, nick: String, email: String, password: String): String =
        "insert into ${tableName()}(name_${tableName()}, nick_${tableName()}, email_${tableName()}, " +
            "password_${tableName()}) values ('$name', '$nick', '$email', md5('$password'));"
    
    override fun tableName(): String = "artist"
}

File: repository/artist/ArtistSQL.py
from repository import BaseSQL as bq


class ArtistSQL(bq.BaseSQL):
    @property
    def table_name(self):
        return "artist"

    def insert_artist(self, name: str, nick: str, photo: str, email: str, password: str) -> str:
        return "insert into %s (name_%s, nick_%s, photo_%s, email_%s, password_%s) values ('%s', " \
               "'%s', '%s', '%s', md5('%s'));" % (
                   self.table_name.upper(), self.table_name, self.table_name, self.table_name,
                   self.table_name, self.table_name, name, nick, photo, email, password)
